neuraxle/base.py

```diff
@@ -811,6 +811,40 @@ def set_hyperparams_space(self, hyperparams_space: HyperparameterSpace) -> 'Base
         self.hyperparams_space = HyperparameterSpace(hyperparams_space).to_flat()
         return self
 
+    def update_hyperparams_space(self, hyperparams_space: HyperparameterSpace) -> 'BaseStep':
+        """
+        Update the step hyperparameter spaces without removing the already-set hyperparameters.
+        This can be useful to add more hyperparameter spaces to the existing ones without flushing the ones that were already set.
+
+        Example :
+
+        .. code-block:: python
+
+            step.set_hyperparams_space(HyperparameterSpace({
+                'learning_rate': LogNormal(0.5, 0.5)
+                'weight_decay': LogNormal(0.001, 0.0005)
+            }))
+
+            step.update_hyperparams_space(HyperparameterSpace({
+                'learning_rate': LogNormal(0.5, 0.1)
+            }))
+
+            assert step.get_hyperparams_space()['learning_rate'] == LogNormal(0.5, 0.1)
+            assert step.get_hyperparams_space()['weight_decay'] == LogNormal(0.001, 0.0005)
+
+        :param hyperparams_space: hyperparameters space
+        :type hyperparams_space: HyperparameterSpace
+        :return: self
+        :rtype: BaseStep
+
+        .. seealso::
+            :func:`~BaseStep.update_hyperparams`,
+            :class:`neuraxle.hyperparams.space.HyperparameterSpace`
+        """
+        self.hyperparams_space.update(hyperparams_space)
+        self.hyperparams_space = HyperparameterSamples(self.hyperparams_space).to_flat()
```

Comment:
what is our default convention? was it flat or nested?

Reply:
I guess your code is right but with this question I just want to make you double check to be sure just in case.